src/controller/CHIPDeviceController.h

```diff
@@ -476,6 +476,17 @@ class DLL_EXPORT DeviceCommissioner : public DeviceController,
      */
     CHIP_ERROR GetAttestationChallenge(ByteSpan & attestationChallenge);
 
+    /**
+     * @brief
+     *   This function returns the attestation challenge for the secure session of the device being commissioned.
+     *
+     * @param[out] attestationChallenge The output for the attestationChallenge
+     * @param[in] proxy The DeviceProxy whose secure session should be used to derive the challenge
+     *
+     * @return CHIP_ERROR               CHIP_NO_ERROR on success, or CHIP_ERROR_INVALID_ARGUMENT if no secure session is active
+     */
+    CHIP_ERROR GetAttestationChallenge(DeviceProxy * proxy, ByteSpan & attestationChallenge);
```

Comment:
This API shouldn't be on device commissioner, since it does not use any device commissioner state.  It should just be on DeviceProxy.

Reply:
You're right, that's much cleaner. Updated, and removed the existing GetAttestationChallenge from CHIPDeviceController as well.